iothub_client/samples/pnp/pnp_simple_thermostat/pnp_simple_thermostat.c

```diff
@@ -0,0 +1,441 @@
+static double g_allTemperatures = DEFAULT_TEMPERATURE_VALUE;
+
+// snprintf format for building getMaxMinReport
+static const char g_minMaxCommandResponseFormat[] = "{ \"maxTemp\": %.2f, \"minTemp\": %.2f, \"avgTemp\": %.2f, \"startTime\": \"%s\", \"endTime\": \"%s\" }";
+
+// Format string for sending temperature telemetry
+static const char g_TemperatureTelemetryBodyFormat[] = "{ \"temperature\":  %.02f }";
+
+// Response description is an optional, human readable message including more information
+// about the setting of the temperature.  Because we accept all temperature requests, we 
+// always indicate a success.  An actual sensor could optionally return information about
+// a temperature being out of range or a mechanical issue on the device on error cases.
+static const char g_temperaturePropertyResponseDescription[] = "success";
+
+//
+// CopyTwinPayloadToString takes the twin payload data, which arrives as a potentially non-NULL terminated string, and creates
+// a new copy of the data with a NULL terminator.  The JSON parser this sample uses, parson, only operates over NULL terminated strings.
+//
+static char* CopyTwinPayloadToString(const unsigned char* payload, size_t size)
+{
+    char* jsonStr;
+
+    if ((jsonStr = (char*)malloc(size+1)) == NULL)
+    {
+        LogError("Unable to allocate %lu size buffer", (unsigned long)(size+1));
+    }
+    else
+    {
+        memcpy(jsonStr, payload, size);
+        jsonStr[size] = 0;
+    }
+
+    return jsonStr;
+}
+
+//
+// BuildMaxMinCommandResponse builds the response to the command for getMaxMinReport
+//
+static bool BuildMaxMinCommandResponse(const char* sinceStr, unsigned char** response, size_t* responseSize)
+{
+    int responseBuilderSize;
+    unsigned char* responseBuilder = NULL;
+    bool result;
+
+    if ((responseBuilderSize = snprintf(NULL, 0, g_minMaxCommandResponseFormat, g_minTemperature, g_maxTemperature, g_allTemperatures / g_numTemperatureUpdates, sinceStr, sinceStr)) < 0)
+    {
+        LogError("snprintf to determine string length for command response failed");
+        result = false;
+    }
+    else if ((responseBuilder = calloc(1, responseBuilderSize + 1)) == NULL)
+    {
+        LogError("Unable to allocate %lu bytes", (unsigned long)(responseBuilderSize + 1));
+        result = false;
+    }
+    else if ((responseBuilderSize = snprintf((char*)responseBuilder, responseBuilderSize + 1, g_minMaxCommandResponseFormat, g_minTemperature, g_maxTemperature, g_allTemperatures / g_numTemperatureUpdates, sinceStr, sinceStr)) < 0)
+    {
+        LogError("snprintf to output buffer for command response");
+        result = false;
+    }
+    else
+    {
+        result = true;
+    }
+
+    if (result == true)
+    {
+        *response = responseBuilder;
+        *responseSize = (size_t)responseBuilderSize;
+        LogInfo("Response=<%s>", (const char*)responseBuilder);
+    }
+    else
+    {
+        free(responseBuilder);
+    }
+
+    return response;    
```

Comment:
>     [](start = 20, length = 4)

Here we should return result?

nit: remove ending spaces

Reply:
Fixing both return & spaces.